python/gui/MaaS/templates/maas/dataset_management.html

```diff
@@ -0,0 +1,797 @@
+                // TODO: consider a warning message for receiving an unexpected selection
+                let nav_div = document.getElementById(all_nav_divs[i]);
+                nav_div.style.display = (selected_div_id === all_nav_divs[i]) ? 'block' : 'none';
+            }
+        }
+
+        function buildDatasetDetailsEmptyDiv() {
+            let details_div = document.createElement('div');
+            details_div.id = 'details-div';
+            details_div.class = 'details-modal';
+            details_div.style.position = 'fixed';
+            details_div.style.zIndex = '1';
+            details_div.style.left = '25%';
+            details_div.style.top = '5%';
+            details_div.style.width = '50%';
+            details_div.style.height = '50%';
+            details_div.style.overflow = 'clip';
+            details_div.style.backgroundColor = '#B7B5B5FF';
+            details_div.style.border = '1px solid #888';
+            details_div.style.padding = '5px';
+            details_div.style.paddingTop = '0px';
+            details_div.style.margin = '15% auto';
+
+            let close_x = document.createElement('span');
+            close_x.style.padding = "0px";
+            close_x.style.cursor = "pointer";
+            close_x.style.fontWeight = "bold";
+            close_x.style.fontSize = "20px";
+            close_x.appendChild(document.createTextNode("\u00d7"));
+            close_x.onclick = function() { details_div.remove() };
+            details_div.appendChild(close_x);
+
+            let details_content_div = document.createElement('div');
+            details_content_div.style.height = '90%';
+            details_content_div.style.overflow = 'auto';
+
+            details_div.appendChild(details_content_div);
+
+            let parent_div = document.getElementById('dataset-manage');
+            parent_div.appendChild(details_div);
+            return details_content_div;
+        }
+
+        function buildDetailsTableRow() {
+            let row = document.createElement('tr');
+            // Skip the first arg, as this is the table itself
+            let colCell;
+            let cellText;
+            for (let i = 1; i < arguments.length; i++) {
+                colCell = document.createElement('th');
+                if (i == 1) {
+                    colCell.style.verticalAlign = 'top';
+                    colCell.style.textAlign = 'right';
+                    colCell.style.paddingRight = '8px';
+                    arguments[i] += ":";
+                }
+                else {
+                    colCell.style.textAlign = 'left';
+                    colCell.style.fontWeight = 'normal';
+                }
+                cellText = document.createTextNode(arguments[i]);
+                colCell.appendChild(cellText);
+                row.appendChild(colCell);
+            }
+            arguments[0].appendChild(row);
+        }
+
+        function buildDatasetFilesEmptyDiv() {
```

Comment:
It might help to define these styles in a css document rather than baking them into the template. They'd be easier to change and help reduce some code bloat.

Reply:
Indeed, I started down that road for several things but didn't get everything and wanted to push this.  I plan to do more clean up, it's just a question of whether it's before or after the review and merge of this.

Reply:
I think we will be able to remove a lot of this if / when we move to `css` `grid` / `flexbox`. I think we are all on the same page though.